best-practices-zarr-ndarray.md

```diff
@@ -0,0 +1,536 @@
+   
+   The `bands` array should contain keys corresponding to data variable names, with values providing band-specific metadata.
+
+3. **CF convention attributes SHOULD be preserved when applicable**
+   
+   For data following CF conventions (Climate and Forecast), relevant attributes like `standard_name`, `units`, and `cell_methods` should be represented in STAC metadata.
+
+   TODO: to be refined with the community.
+
+## Required Extensions
+
+For Zarr assets, the following extensions are typically required:
+
+- **Datacube Extension (v2.x)**: To describe multidimensional structure
+  - `cube:dimensions`: Define coordinate dimensions
+  - `cube:variables`: Describe data variables within groups
+
+- **Projection Extension (v2.x)**: For georeferenced arrays
+  - `proj:code`, `proj:wkt2`, or `proj:projjson`
+  - `proj:shape` and `proj:transform`
+
+- **Raster Extension (v2.x)**: For raster-like arrays
+  - `raster:spatial_resolution`
+  - `raster:scale` and `raster:offset` (if applicable)
+  - `data_type` and `nodata`
+
+## Use Cases
+
+### Scene-Based Zarr Stores (EOPF Products)
+
+**Context**: Earth Observation Processing Framework (EOPF) products organize data in hierarchical Zarr stores with multiple resolution groups. Example: Sentinel-2 Level-2A data.
+
+**Store Structure**:
+```
+/
+├── conditions/
+├── measurements/
+│   └── reflectance/     
+│       ├── r10m/
+│       │   ├── b02 (10980, 10980) uint16
+│       │   ├── b03 (10980, 10980) uint16
+│       │   ├── b04 (10980, 10980) uint16
+│       │   ├── b08 (10980, 10980) uint16
+│       │   ├── x (10980,) int64
+│       │   └── y (10980,) int64
+│       ├── r20m/
+│       └── r60m/   
+└── quality/
+```
+
+**STAC Representation**:
+
+```json
+{
+  "type": "Feature",
+  "stac_version": "1.1.0",
+  "id": "S2B_MSIL2A_20251006T123309",
+  "properties": {
+    "datetime": "2025-10-06T12:33:09Z"
+  },
+  "links": [
+    {
+      "rel": "store",
+      "href": "s3://bucket/S2B_MSIL2A_20251006T123309.zarr",
+      "type": "application/vnd+zarr; version=2",
+      "title": "Sentinel-2 L2A Zarr Store"
+    }
+  ],
+  "assets": {
+    "measurements_r10m": {
+      "href": "s3://bucket/S2B_MSIL2A_20251006T123309.zarr/measurements/reflectance/r10m",
+      "type": "application/vnd+zarr; version=2",
+      "roles": ["group", "data"],
+      "raster:spatial_resolution": 10,
+      "data_type": "uint16",
+      "nodata": 0,
+      "raster:scale": 0.0001,
+      "raster:offset": 0.0,
+      "proj:code": "EPSG:32628",
+      "proj:shape": [10980, 10980],
+      "cube:dimensions": {
+        "x": {
+          "type": "spatial",
+          "axis": "x",
+          "extent": [300000, 409800]
+        },
+        "y": {
+          "type": "spatial",
+          "axis": "y",
+          "extent": [0, 109800]
+        }
+      },
+      "cube:variables": {
+        "b02": {
+          "dimensions": ["y", "x"],
+          "type": "data",
+          "unit": null
```

Comment:
`unit` is not a mandatory field of the datacube extension, should we still keep it here?

Reply:
I think the idea was to differentiate between the two cases:

- this variable is known to be unitless: `"unit": null
- unit information just wasn't included, so it could be anything